homeassistant/components/vesync/diagnostics.py

```diff
@@ -0,0 +1,141 @@
+"""Diagnostics support for VeSync."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyvesync import VeSync
+
+from homeassistant.components.diagnostics import REDACTED
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .common import VeSyncBaseDevice
+from .const import DOMAIN, VS_MANAGER
+
+KEYS_TO_REDACT = {"manager"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    manager: VeSync = hass.data[DOMAIN][VS_MANAGER]
+
+    data = {}
```

Comment:
You can either define 1 dictionary at once

```python
data = {
  DOMAIN: { ... }
}
```

or if you prefer to assign 1 key at a time, do that instead of using the `update` method:

```python
data[DOMAIN] = { ... }
```

Reply:
Changed to 1 dictionary